### DISCUSSION TITLE
Error when trying to read data from K30 CO2 sensor

### DISCUSSION THREAD

**Discussion by alexnaoki** [micropython/micropython#12401] [RP2040 / Pico]
I am new to I2C, but I've been trying to use the K30 CO2 sensor from Senseair, and I've been trying to follow one of their links regarding [I2C communication guide](https://rmtplusstoragesenseair.blob.core.windows.net/docs/Dev/publicerat/TDE4700.pdf). 

I tried to follow the steps from page 28 from the link above, "Appendix B, I2C bus transaction examples":

However, I keep getting the error 
```py
Traceback (most recent call last):
  File "<stdin>", line 16, in <module>
OSError: [Errno 5] EIO
```
My code below: 

```py
from machine import Pin, I2C
import time

i2c = I2C(0, scl=Pin(1), sda=Pin(0))
print(i2c.scan())

address = 0x68  #default sensor address
CMD_READ_REG = 0x22
REG_CO2_PPM = 0x08
sensor_adress_shift = address << 1
checksum = (CMD_READ_REG+REG_CO2_PPM) & 0xFF

order = [sensor_adress_shift, CMD_READ_REG, 0x00, REG_CO2_PPM, checksum]
buffer = bytes(order)

i2c.writeto_mem(address, sensor_adress_shift, buffer, addrsize=8)
time.sleep_ms(30)

response = i2c.readfrom(address, 4)
print(response)

```

**Comment by uGinn-R** [comment]
Your i2c.writeto_mem() format is incorrect The documentation show this: `I2C.writeto_mem(addr, memaddr, buf, *, addrsize=8)`.
So you have to provide (DEVICE_ADDRESS, MEMORY_REGISTER, DATA_TO_WRITE). You're a trying to pass bitshifted device address as a memory register. Also pay attention to your payload called "order", as far as i understood from datasheet the order should looks like this [0x22, 0x00, 0x08, 0x2A]
